pkg/webservice/handlers.go

```diff
@@ -291,3 +293,33 @@ func getContainerHistory(w http.ResponseWriter, r *http.Request) {
 	}
 
 }
+
+func getClusterConfig(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+	jsonOutput := false
+	query := r.URL.Query()
+	// check if we have a request for json output
+	if len(query) != 0 {
+		if _, ok := query["json"]; ok {
+			jsonOutput = true
+		}
+	}
+
+	var marshalledConf []byte
+	var err error
+	conf := configs.ConfigContext.Get(gClusterInfo.GetPolicyGroup())
+	if jsonOutput {
+		marshalledConf, err = json.Marshal(&conf)
+	} else {
+		marshalledConf, err = yaml.Marshal(&conf)
```

Comment:
I am not sure if yaml is a valid response format for a RESTful API. 
I think json is good enough because the rest of our API all return json. Shall we keep consistency?

Reply:
REST does not specify, require or limit anything we return it can be any text form as well as binary.

We should provide yaml as that is the format the original file is in. We should not expect that the client **must** implement a json to yaml conversion tool before it can use the file. It might also lead to data issues.

Reply:
I think its good to see the default the response as yaml.
As I see, its not good to compare headers like r.Header.Get. If possible, could we some how have a cleaner interfaces to find/get some header if present. Some time, comparison can have issues in unicode, caps etc etc. It's better we have a util package/class to handle some checks in a standard and use some bool/enum to handle the further processing.

Reply:
The header retrieval is hidden inside the go packages. It already handles the unicode and caps etc. Predefined MIME headers should however only be ascii, specially for the ones we use. This is from the comments in the go source code:
```
// Get gets the first value associated with the given key. If
// there are no values associated with the key, Get returns "".
// It is case insensitive; textproto.CanonicalMIMEHeaderKey is
// used to canonicalize the provided key. To access multiple
// values of a key, or to use non-canonical keys, access the
// map directly.
```
So we're covered unless we come up with some special non standard headers.